DataViews/VerbosityHeaderHandler.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace DataViews
+{
+    public class VerbosityHeaderHandler : DelegatingHandler
+    {
+        public VerbosityHeaderHandler(bool verbose = true)
+        {
+            Verbose = verbose;
+        }
+
+        public bool Verbose { get; set; }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            // If the handler is set to non-verbose, set the accept-verbosity header to non-verbose to prevent null values from being returned from OCS
+            if (!Verbose)
+            {
+                request?.Headers.Add("accept-verbosity", "non-verbose");
```

Comment:
Request is already being null checked above so no need for this coalescing. 

That said, request will never be null in a delegating handler anyway so you don't really need to check it, there is a compiler warning about this (CA1508) that we have disabled. I don't mind keeping it to be on the safe side though, just an FYI.

Reply:
I just realized that CA1062 will throw without a null check, so nevermind on the second point :)

Reply:
this also could be in the else branch and then you don't have to redo the check